packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -269,7 +269,7 @@ class Localizations extends StatefulWidget {
   static Locale localeOf(BuildContext context) {
     assert(context != null);
     final _LocalizationsScope scope = context.inheritFromWidgetOfExactType(_LocalizationsScope);
-    return scope.localizationsState.locale;
+    return scope?.localizationsState?.locale ?? window.locale;
```

Comment:
Past experience has suggested that it's better if the call sites do the defaulting rather than the .of functions. It's ok to return null if there's no _LocalizationsScope in scope (though then you should document that that will happen), but we shouldn't default.

The problem with defaulting is that people don't know the value came from `window`, so they don't think to set up the onLocaleChanged handler. We saw this a lot with MediaQuery.of defaulting its padding to `window.padding` if it couldn't find a MediaQuery.